enterprise-suite/scripts/lbc.py

```diff
@@ -284,13 +293,91 @@ def check_resource_list(cmd, expected, fail_msg):
         return all_found
     return False
 
-# Checks for console PVCs
-def are_pvcs_created(namespace):
-    return check_resource_list(
-        cmd='kubectl get pvc --namespace={} --no-headers'.format(namespace),
-        expected=CONSOLE_PVCS,
-        fail_msg='Found some PVCs ({}) from previous console install, but not all expected: {}.\nTo avoid data loss, please remove them manually'
-    )
+# Returns two lists.  For each of the PVC types we care about, the associated PV
+# is in the first or second list if the reclaim policy is RETAIN or not respectively.
+# Each list element is a tuple of PV name and claim.
+def pvsRetainedOrNot(namespace):
+    returncode, stdout, _ = run('kubectl --namespace {} get pv --no-headers'
+                             .format(namespace))
+    retained = []
+    notRetained = []
+
+    if returncode != 0:
+        printerr("Unable to retrieve PV info.  Proceed with caution.")
+    else:
+        for pvc in CONSOLE_PVCS:
+            # We might want to do this by parsing yaml instead...
+            # pvc-accccc27-2ef8-11e9-a408-080027c68b7b   32Gi       RWO            Delete           Bound    lightbend/alertmanager-storage   standard                24h
+            claim = namespace + '/' + pvc
+
+            match = re.search(r'^([\w-]+)\s+\w+\s+\w+\s+(\w+)\s+Bound\s+('+claim+r')\s', stdout, re.MULTILINE)
+
+            if match:
+                if match.group(2) == 'Retain':
+                    list = retained
+                else:
+                    list = notRetained
+                list.append((match.group(1), match.group(3)))
+
+    return (retained, notRetained)
+
+# This is the behavior as described at
+# https://github.com/lightbend/es-backend/issues/572
+def checkPVthings(aboutToUninstall=False, namespace=None):
+    returncode, stdout, stderr = run('helm get ' + args.helm_name,
+                            DEFAULT_TIMEOUT, show_stderr=True)
+
+    if (returncode != 0) and not ('Error: release: "{}" not found'.format(args.helm_name) in stderr):
+        printerr("Unable to retrieve PV info.  Proceed with caution.")
+        return
+
+    hasPVs = 'usePersistentVolumes: true' in stdout
+
+    # This assumes the default is true.  What happens if they modify this in values.yaml?  We'll miss that I think.
+    wantsPVs = len(filter(lambda x: 'usePersistentVolumes=false' in x, sys.argv)) == 0
```

Comment:
This pattern is coming up in a few places, let's extract it into a function `get_args_helm_value(name)` or similar. Then we can also later improve it to work better with computed or values.yaml stuff if needed.

Reply:
This doesn't seem like a bomb proof approach either. It would be much better if we can check the actual parsed arg value - I thought we already use getopts?

Reply:
Hmmm.  The existing `lambda` code does check the parsed arg value as I understand it (unless you meant matching on `false`, `False`, `0` etc.).  My comment was that this will miss the case where the user changes the value (e.g. `usePersistentVolumes`) in the `values.yaml` (or any such values file) referred to in the helm-specific values passed to `lbc.py`.  For example, instead of specifying

`./lbc.py install --local-chart=.. -- --set usePersistentVolumes=true`
the user does something like
`./lbc.py install --local-chart=.. -- -f setsUsePersistentVolume.yaml`   or the very unlikely (imho)
`./lbc.py install --local-chart=.. -- --set-file usePersistentVolumes=file/with/true`
To be completely general, we'd also have to deal with
`./lbc.py install --local-chart=.. -- --set-string someStringVarWeCareAbout=57`

And now that I think about it, I wonder if we deal properly with the "rightmost value wins" semantics used by helm.

In any case, I just went with the flow for now...